core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -92,6 +92,9 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   // The num of current max ExecutorId used to re-register appMaster
   @volatile protected var currentExecutorIdCounter = 0
 
+  // The set of executors we have on each host.
+  protected val hostToExecutors = new HashMap[String, HashSet[String]]
```

Comment:
`CoarseGrainedSchedulerBackend` already has a `TaskSchedulerImpl`, which has this functionality in `getExecutorsAliveOnHost`.  I think you should just call that.